### PR TITLE
fix(misconf): Escape template value correctly

### PR DESCRIPTION
**Pull request by simar7**
## Description

Escape templates correctly.

## Related issues
- Close https://github.com/aquasecurity/trivy/issues/6291

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/iac/terraform/resource_block.go [hunk 1]
```diff
@@ -108,6 +103,24 @@ EOF
 
 }
 
+func parseStringPrimitive(input string) string {
+	// we must escape templating
+	// ref: https://developer.hashicorp.com/terraform/language/expressions/strings#escape-sequences-1
+	r := regexp.MustCompile(`((\$|\%)\{.+\})`)
+	ff := r.ReplaceAllStringFunc(input, func(s string) string {
+		s = strings.ReplaceAll(s, "$", "$$")
+		s = strings.ReplaceAll(s, "%", "%%")
```
Comment (nikpivkin) [inline pkg/iac/terraform/resource_block.go:112]: Only the first `$` or `%` character should be escaped, as this character may occur in the string. For example: `${test$} -> $${test$}`

**Comment by nikpivkin** [comment]
LGTM, but left a couple of small comments

File: pkg/dependency/parser/golang/mod/testdata/replaced-with-local-path-and-version-mismatch/go.sum [hunk 2]
```diff
@@ -50,6 +50,7 @@ golang.org/x/tools v0.0.0-20190621195816-6e04913cbbac/go.mod h1:/rFqwRUd4F7ZHNgw
 golang.org/x/tools v0.0.0-20191029041327-9cc4af7d6b2c/go.mod h1:b+2E5dAYhXwXZwtnZ6UAqBI28+e2cm9otk0dWdXHAEo=
 golang.org/x/tools v0.0.0-20191029190741-b9c20aec41a5/go.mod h1:b+2E5dAYhXwXZwtnZ6UAqBI28+e2cm9otk0dWdXHAEo=
 golang.org/x/xerrors v0.0.0-20190717185122-a985d3407aa7/go.mod h1:I/5z698sn9Ka8TeJc9MKroUUfqBBauWjQqLJ2OPfmY0=
+golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1/go.mod h1:I/5z698sn9Ka8TeJc9MKroUUfqBBauWjQqLJ2OPfmY0=
```
Comment (nikpivkin) [inline pkg/dependency/parser/golang/mod/testdata/replaced-with-local-path-and-version-mismatch/go.sum:53]: Why is this file changed?

File: pkg/iac/scanners/terraformplan/tfjson/test/testdata/plan_with_template.json [hunk 3]
```diff
@@ -0,0 +1,480 @@
+{
```
Comment (nikpivkin) [inline pkg/iac/scanners/terraformplan/tfjson/test/testdata/plan_with_template.json:1]: This file can be minified

**Review by nikpivkin — APPROVED** [review]
LGTM, but left a couple of small comments

File: pkg/dependency/parser/golang/mod/testdata/replaced-with-local-path-and-version-mismatch/go.sum (hunk 2 above)
Comment (simar7) [reply pkg/dependency/parser/golang/mod/testdata/replaced-with-local-path-and-version-mismatch/go.sum:53]: accidental change, will remove.

File: pkg/iac/scanners/terraformplan/tfjson/test/testdata/plan_with_template.json (hunk 3 above)
Comment (simar7) [reply pkg/iac/scanners/terraformplan/tfjson/test/testdata/plan_with_template.json:1]: Yes but it's nice to be able to read the test case input if any debugging is needed.

Comment (nikpivkin) [reply pkg/iac/scanners/terraformplan/tfjson/test/testdata/plan_with_template.json:1]: I agree
